pkg/eventemitter/azure_event_grid_topic_handler.go

```diff
@@ -0,0 +1,133 @@
+
+	eventingv1alpha1 "github.com/kedacore/keda/v2/apis/eventing/v1alpha1"
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+	"github.com/kedacore/keda/v2/pkg/eventemitter/eventdata"
+	"github.com/kedacore/keda/v2/pkg/scalers/azure"
+)
+
+type AzureEventGridTopicHandler struct {
+	Context      context.Context
+	Endpoint     string
+	ClusterName  string
+	Client       *publisher.Client
+	logger       logr.Logger
+	activeStatus metav1.ConditionStatus
+}
+
+func NewAzureEventGridTopicHandler(context context.Context, clusterName string, spec *eventingv1alpha1.AzureEventGridTopicSpec, authParams map[string]string, podIdentity kedav1alpha1.AuthPodIdentity, logger logr.Logger) (*AzureEventGridTopicHandler, error) {
+	var err error
+	var client *publisher.Client
+
+	switch podIdentity.Provider {
+	case "", kedav1alpha1.PodIdentityProviderNone:
+		if authParams["key"] == "" {
+			err = fmt.Errorf("no key provided")
+			break
+		}
+		client, err = publisher.NewClientWithSharedKeyCredential(spec.EndPoint, azcore.NewKeyCredential(authParams["key"]), nil)
+	case kedav1alpha1.PodIdentityProviderAzure, kedav1alpha1.PodIdentityProviderAzureWorkload:
+		creds, chainedErr := azure.NewChainedCredential(logger, podIdentity.GetIdentityID(), podIdentity.GetIdentityTenantID(), podIdentity.Provider)
+		if chainedErr != nil {
+			err = chainedErr
+			break
+		}
+		client, err = publisher.NewClient(spec.EndPoint, creds, nil)
+	default:
+		err = fmt.Errorf("incorrect auth provided")
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	logger.Info("Create new azure event grid handler")
+	return &AzureEventGridTopicHandler{
+		Context:      context,
+		Client:       client,
+		Endpoint:     spec.EndPoint,
+		ClusterName:  clusterName,
+		logger:       logger,
+		activeStatus: metav1.ConditionTrue,
+	}, nil
+}
+
+func (a *AzureEventGridTopicHandler) CloseHandler() {
+
+}
+
+func (a *AzureEventGridTopicHandler) SetActiveStatus(status metav1.ConditionStatus) {
+	a.activeStatus = status
+}
+
+func (a *AzureEventGridTopicHandler) GetActiveStatus() metav1.ConditionStatus {
+	return a.activeStatus
+}
+
+func (a *AzureEventGridTopicHandler) EmitEvent(eventData eventdata.EventData, failureFunc func(eventData eventdata.EventData, err error)) {
+	type emitData struct {
+		Reason  string `json:"reason"`
+		Message string `json:"message"`
+	}
+
+	source := fmt.Sprintf("/%s/%s/keda", a.ClusterName, kedaNamespace)
+	subject := fmt.Sprintf("/%s/%s/%s/%s", a.ClusterName, eventData.Namespace, eventData.ObjectType, eventData.ObjectName)
+
+	opt := &messaging.CloudEventOptions{
+		Subject:         &subject,
+		DataContentType: to.Ptr("application/json"),
+		Time:            &eventData.Time,
+	}
+
+	event, err := messaging.NewCloudEvent(source, eventData.EventType, emitData{Reason: eventData.Reason, Message: eventData.Message}, opt)
+
+	if err != nil {
+		a.logger.Error(err, "EmitEvent error %s")
+		return
+	}
+
+	eventsToSend := []messaging.CloudEvent{
+		event,
+	}
```

Comment:
Any reason why we are building the event here as well and not re-using what we are already doing?